### DISCUSSION TITLE
Password reset link fragments disappearing when set to localhost

### DISCUSSION THREAD

**Discussion by brycegoh** [supabase/supabase#2176] [Questions]
Hello, I am currently having difficulty testing the reset password functionality of my app. When ever supabase send a password reset email with the link, the link resolves to empty fragments when clicked such as http://localhost:3000/auth#. Would appreciate some help with this. Thank you!!

**Comment by awalias** [comment]
hey @brycegoh ! Could you let us know:
- what you have as the Site Url in the auth settings
- if you have any 'Additional URLs' in settings
- whether you are setting `redirectTo` param on the client library
- which library or framework you are using

cheers!

**Reply by brycegoh** [reply]
Hi, thanks for the follow up.

1. What you have as the Site Url in the auth settings
site url is *http://localhost:3000/auth*

2. Any 'Additional URLs' in settings
No additional URL in settings

3. Using react with react router.

### front end:
```javascript
function App() {
  return (
      <BrowserRouter>
      <Switch>
          <Route exact path="/auth" component={Auth} />
      </Switch>
    </BrowserRouter>
  );
}
```
### Auth Component
```javascript
const Auth = () => {

  const urlType = useFragmentString("type");
  const errorCode = useFragmentString("error_code");

  if (errorCode === "404") {
    return ExpiredLink();
  } else if (urlType === "signup") {
    return EmailVerified();
  } else if (urlType === "recovery") {
    return ForgetPassword();
  }
};
```

### Extracting fragments
```javascript
import qs from 'qs';

const getFragmentStringValue = ( 
    key, 
    hashString = window.location.hash.substring(1),
) => { 
    const values = qs.parse(hashString); 
    return values[key] ? values[key] : null;
};

export const useFragmentString = ( key ) => {
    return getFragmentStringValue(key)
}
```

edit: Redirect to ExpiredLink Component works when trying to re-access the emailed link. Seems like it is only happening for redirect to ForgotPassword Component.

**Reply by brycegoh** [reply]
Hi, sorry to trouble but any updates on this? I am still getting this bug even when not on localhost.
